File: ScriptureFeed.Web/Models/Verse.cs
using System;
using System.Collections.Generic;

namespace ScriptureFeed.Web.Models
{
    [Serializable]
    public class Verse
    {
        public string Id { get; set; }
        public string VerseText { get; set; }
        public string ImageLink { get; set; }
        public DateTime VerseDate { get; set; }
        public string VideoLink { get; set; }
        public string ReferenceLink { get; set; }
        public string VerseNumbers { get; set; }
        public string Chapter { get; set; }
        public string Book { get; set; }
        public string ReferenceText { get; set; }
        public string BibleReferenceLink { get; set; }
        public string FacebookShareUrl { get; set; }
        public string TwitterShareUrl { get; set; }
        public string PinterestShareUrl { get; set; }
        public string Url { get; set; }
        public bool IsFavorite { get; set; } = false;
    }

    [Serializable]
    public class GetVerseResponse
    {
        public List<Verse> Verses { get; set; }
        public int PageSize { get; set; }
        public int PageNumber { get; set; }
        public bool HasMorePages { get; set; }
    }
}

File: ScriptureFeed.Web/ClientApp/src/app/models/verse.ts
export class Verse implements IVerse {
  public id: string | '';
  public verseText: string | '';
  public imageLink: string | '';
  public book: string | '';
  public chapter: string | '';
  public verseNumbers: string | '';
  public isFavorite: boolean | false;

  constructor() { }

  getCitationInfo(): string {
    let returnVal = `${this.book} ${this.chapter}:${this.verseNumbers}`;
    return returnVal;
  }
}

export interface IVerse {
  id: string;
  verseText: string;
  imageLink: string;
  book: string;
  chapter: string;
  verseNumbers: string;
  isFavorite: boolean;
  getCitationInfo(): string;
}

export interface IGetVerseResponse {
  verses: Array<Verse>,
  pageSize: number,
  pageNumber: number,
  hasMorePages: boolean
}
